docs/pages/docs/routing/middleware.mdx

```diff
@@ -549,29 +549,113 @@ export const config = {
 
 ### Example: Integrating with Supabase Authentication
 
-[`@supabase/auth-helpers-nextjs`](https://supabase.com/docs/guides/auth/auth-helpers/nextjs?language=ts#managing-session-with-middleware) exports `createMiddlewareClient` that accepts a `req` and `res` object. We can pass the incoming `req` and the `res` created with `createIntlMiddleware` to it.
+[`@supabase/ssr`](https://supabase.com/docs/guides/auth/server-side/creating-a-client) exports a `createServerClient` function. First we're going to create a utils function that invokes the `createServerClient` function and returns the `supabase` object.
+
+```ts filename="utils/supabase/middleware.ts"
+import {createServerClient, type CookieOptions} from '@supabase/ssr';
+import {type NextRequest, NextResponse} from 'next/server';
+
+export const createClient = (request: NextRequest) => {
+  // Create an unmodified response
+  let response = NextResponse.next({
```

Comment:
This shouldn't be necessary, we should use the `response` from `handleI18nRouting`.

Reply:
refactored accordingly